cpu/cpustress.py

```diff
@@ -0,0 +1,307 @@
+    multiple cycles of cpu toggle
+
+    1. serial off all, serial on all cpus
+    2. off/on single cpu 100 times all cpus
+    3. off/on one cpu at a time all cpus
+    4. toggle first off and second half cpus on
+    5. Affine task to single cpu and do off on check for the process (init)
+    6. affine to shared multiple cpus and off on (sleep)
+    7. Do multiple cpu off on at once ppc64_cpu --smt
+    8. LPAR/GUEST : trigger DLPAR CPU using drmgr command
+
+    :avocado: tags=cpu,power,privileged
+    """
+
+    def setUp(self):
+        """
+        Check required packages is installed, and get current SMT value.
+        """
+        sm = SoftwareManager()
+        pkgs = ['util-linux', 'powerpc-utils', 'numactl']
+        self.curr_smt = process.system_output(
+            "ppc64_cpu --smt | awk -F'=' '{print $NF}' | awk '{print $NF}'", shell=True)
+        if 'ppc' not in platform.processor():
+            self.cancel("Processor is not powerpc")
+        for pkg in pkgs:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.cancel("%s is required to continue..." % pkg)
+
+    @staticmethod
+    def __clear_dmesg():
+        process.run("dmesg -c", sudo=True)
+
+    @staticmethod
+    def __error_check():
+        logs = process.system_output(
+            "dmesg -T --level=alert,crit,err,warn", ignore_status=True, shell=True, sudo=True)
+        for error in errorlog:
+            if any(error in log for log in logs.split()):
+                ERROR = 'Test resulted %s in syslogs"' % error
+                return ERROR
+
+    @staticmethod
+    def __isSMT():
+        if 'is not SMT capable' in process.system_output("ppc64_cpu --smt"):
+            return False
+        return True
+
+    @staticmethod
+    def __online_cpus(cores):
+        for cpus in range(cores):
+            cpu.online(cpus)
+
+    @staticmethod
+    def __offline_cpus(cores):
+        for cpus in range(cores):
+            cpu.ffline(cpus)
+
+    @staticmethod
+    def __cpu_toggle(core):
+        if cpu._get_cpu_status(core):
+            cpu.offline(core)
+        else:
+            cpu.online(core)
+
+    @staticmethod
+    def __kill_process(pids):
+        for pid in pids:
+            process.run("kill -9 %s" % pid)
+
+    def test(self):
+        """
+        calls each of the test in a loop for the given values
+        """
+        iteration = self.params.get('iteration', default='1')
+        timeout = self.params.get('timeout', default='1')
+        tests = self.params.get('test', default='all')
+        self.__online_cpus(totalcpus)
+        if 'all'in tests:
+            tests = ['cpu_serial_off_on',
+                     'single_cpu_toggle',
+                     'cpu_toggle_one_by_one',
+                     'multiple_cpus_toggle',
+                     'pinned_cpu_stress',
+                     'dlpar_cpu_hotplug']
+        else:
+            tests = tests.split()
```

Comment:
if 'all' is passed from yaml, the for loop go over the list tests.
but if single test given, than its a string so make it a list tests.split()